pkg/providers/vsphere/vsphere_test.go

```diff
@@ -2004,6 +2004,54 @@ func TestSetupAndValidateCreateClusterEtcdMachineGroupRefNonexistent(t *testing.
 	}
 }
 
+func TestSetupAndValidateCreateClusterOsFamilyInvalid(t *testing.T) {
+	ctx := context.Background()
+	clusterSpec := givenEmptyClusterSpec()
+	fillClusterSpecWithClusterConfig(clusterSpec, givenClusterConfig(t, testClusterConfigMainFilename))
+	provider := givenProvider(t)
+	controlPlaneMachineConfigName := clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
+	provider.machineConfigs[controlPlaneMachineConfigName].Spec.OSFamily = "rhel"
+	var tctx testContext
+	tctx.SaveContext()
+
+	err := provider.SetupAndValidateCreateCluster(ctx, clusterSpec)
+	if err != nil {
```

Comment:
Wondering if we should remove the `err != nil` check
If the method returns nil, the test should fail, right?

Reply:
yeah you are right. I copied it from one of the other tests but it's prolly not needed their either